lib/cfa/augeas_parser/writer.rb

```diff
@@ -293,7 +293,15 @@ def insert_entry(located_entry)
       # @param located_entry [LocatedEntry] entry to insert
       # @return [String] where value should be written.
       def insert_after(preceding, located_entry)
-        aug.insert(preceding.path, located_entry.key, false)
+        res = aug.insert(preceding.path, located_entry.key, false)
+        # if insert failed it means, that previous preceding entry was single
+        # element and now it is multiple ones, so try to first element as add
+        # is done in reverse order
+        if res == -1
+          # TODO: what about deep nesting of trees where upper level change from single
+          # to collection?
+          res = aug.insert(preceding.path+"[1]", located_entry.key, false)
```

Comment:
NP: Rubocop would feel offended because `Layout/SpaceAroundOperators` :)

Reply:
Although seem to be offended because `Layout/HashAlignment` https://travis-ci.org/config-files-api/config_files_api/jobs/630915224#L756